contrib/parseq-restli-client/src/main/java/com/linkedin/restli/client/ParSeqRestClient.java

```diff
@@ -194,13 +207,56 @@ static String generateTaskName(final Request<?> request) {
     }
   }
 
+  private <T> Task<Response<T>> withD2Timeout(final Task<Response<T>> task, ConfigValue<Long> timeout) {
+    if (timeout.getValue() != null && timeout.getValue() > 0) {
+      // transform TimeoutException to be compatible with current Task timeout exception message
+      String desc = timeout.getSource().map(src -> " src: " + src).orElse("");
+      String timeoutTaskName = "withTimeout " + Math.min(timeout.getValue(), Integer.MAX_VALUE)
```

Comment:
The task name does not take into consideration reconciled value from RequestContext. Ideally we could include reconciled value and D2 as source if it was the source of a timeout.

Reply:
Make sense. Let me enhance this.